crates/matrix-sdk/src/room/joined.rs

```diff
@@ -825,6 +829,38 @@ impl Joined {
         self.send(RoomMessageEventContent::new(content), config.txn_id.as_deref()).await
     }
 
+    /// Update the power levels of a select set of users of this room.
+    ///
+    /// Issue a `power_levels` state event request to the server, changing the
+    /// given user_id -> i64 levels. May fail if the `power_levels` aren't
+    /// locally known yet or the server rejects the state event update, e.g.
+    /// because of insufficient permissions. Neither permissions to update
+    /// nor whether the data might be stale is checked prior to issuing the
+    /// request.
+    pub async fn update_power_levels(
+        &self,
+        updates: Vec<(&UserId, Int)>,
+    ) -> Result<send_state_event::v3::Response> {
+        let Some(current_powers) : Option<SyncRoomPowerLevelsEvent>= self
+            .get_state_event_static()
+            .await?
+            .and_then(|e| e.deserialize().ok()) else {
+                return Err(crate::Error::InsufficientData)
+            };
```

Comment:
We shouldn't return `InsufficientData` if deserialization fails.

```suggestion
        let raw_pl_event = self
            .get_state_event_static::<RoomPowerLevelsEventContent>()
            .ok_or(Error::InsufficientData)?;
        let pl_event = r.deserialize()?;
```